src/arrayMethodSort.js

```diff
@@ -4,9 +4,26 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = (a, b) => `${a}` > `${b}`) {
+    for (let i = 0; i < this.length - 1; i++) {
+      if (
+        compareFunction(this[i], this[i + 1]) === true ||
+        compareFunction(this[i], this[i + 1]) > 0
```

Comment:
The condition checks the result of the compare function twice. This is inefficient as the same comparison is being done twice. You can store the result of the compare function in a variable and use that in the if-statement to improve performance.